spec/index.html

```diff
@@ -684,12 +687,18 @@ <h2>Literals</h2>
         language tag MUST be well-formed according to
         <a data-cite="bcp47#section-2.2.9">section 2.2.9</a>
         of [[!BCP47]].</li>
+      <li>if and only if the <a>datatype IRI</a> is
+        <code>http://www.w3.org/1999/02/22-rdf-syntax-ns#langString</code>,
+        <dfn>text direction</dfn> MUST be empty, `ltr` or `rtl`.</li>
```

Comment:
A literal is described as having two to four elements, and there are constraints on when it can have a language tag (or text direction). If there is an additional datatype `rdf:dirLangString` then we could say that language-tagged strings have three components and directional language-tagged-strings have four components, but at some additional cognitive load. If it is all `rdf:langString`, then there are four components, and text direction can have three possible values, empty (""), `ltr`, or `rtl`. SPARQL `LANGDIR` would return any of these three values for a language-tagged string.

Reply:
The cognitive load is introducing a new concept.

`rdf:langString` is currently three slots.

There isn't a datatype class of all strings with a base direction if it is a new facet of `rdf:langString`.

SPARQL, or any toolkit API, can present this in a context suitable way but let's make the core data model exact.

Reply:
I don't believe it is transparent as implied by "less cognitive burden".

Simply ignoring direction changes cardinality but RDF graphs are sets (the usual effect of projection).

Used with `skos:prefLabel` there is a pattern of one label per language tag:
https://www.w3.org/TR/skos-reference/#S14

Finding literals with datatype `rdf:langString` may now find multiple instances if searching on `LANGMATCHES` or `DATATYPE`.